Author: Mattias Ellert <mattias.ellert@fysast.uu.se>
Description: adapt parameters of hdb_generate_key_set_password() to heimdal 1.6~git20120311
 .
 With version heimdal 1.6~git20120311 heimdal schanged the number of parameters
 of function hdb_generate_key_set_password(), implementing a fallback to "default"
 values when NULL-values are passed for these parameters.
 .
 This patch does exactly that.
 .
Bug-Debian: 664930
Reviewed-by: Peter Marschall <peter@adpm.de>

--- openldap-2.4.28.orig/contrib/slapd-modules/smbk5pwd/smbk5pwd.c	2011-11-25 19:52:29.000000000 +0100
+++ openldap-2.4.28/contrib/slapd-modules/smbk5pwd/smbk5pwd.c	2012-03-30 10:03:14.035984880 +0200
@@ -468,7 +468,7 @@
 		}
 
 		ret = hdb_generate_key_set_password(context, ent.principal,
-			qpw->rs_new.bv_val, &ent.keys.val, &nkeys);
+			qpw->rs_new.bv_val, NULL, 0, &ent.keys.val, &nkeys);
 		ent.keys.len = nkeys;
 		hdb_seal_keys(context, db, &ent);
 		krb5_free_principal( context, ent.principal );
